### PR TITLE
feat(api): add jitter to cron jobs, paging support, and backoff handling

### PR DESCRIPTION
**Pull request by IceHamster**
Hello, Zaptec devs are here!

We see some issues with the current implementation of the app from our side (for example scheduled fetch of the data is failing).

We would like to help with a reliability of the plugin for Homey users, and to do so, the following changes are proposed:

* Random jitter to cron jobs to reduce simultaneous endpoint querying — this will help with 4xx and 5xx errors users are experiencing (at scale it looks like mini DoS and will be rate limited heavily).
* Implemented paging support for charging history API calls — this will help to get better throughput at scale.
* Introduced backoff mechanism for handling 429 Too Many Requests responses.
* Added custom User-Agent header to API requests for better tracking — in the future empty user agent may prevent Homey users from using the API.
* Refactored device initialization to include app version.

These changes enhance API robustness and prevent potential overloading while ensuring compatibility with paginated responses. We hope that these changes will be beneficial for all Homey users and will improve the overall experience with the Zaptec app.

We deeply appreciate the time and expertise that go into maintaining and improving the Homey plugin, and we're excited to do a small contribution to its continued success.

### CODE REVIEW

File: drivers/go/device.ts [hunk 1]
```diff
@@ -36,7 +37,13 @@ export class GoCharger extends Homey.Device {
     this.cronTasks.push(
       cron.schedule('0,30 * * * * *', () => this.pollValues()),
```
Comment (philipostli) [inline drivers/go/device.ts:38]: Should there be a jitter on pollVales too? Run every whole and half minute. Same for all users I think.

File: drivers/go/device.ts [hunk 2]
```diff
@@ -257,11 +265,11 @@ export class GoCharger extends Homey.Device {
         ChargerId: this.getData().id,
         From: new Date(year, 0, 1, 0, 0, 1).toJSON(),
         DetailLevel: 0,
-        PageSize: 5000,
+        PageSize: 50,
```
Comment (philipostli) [inline drivers/go/device.ts:268]: Good. This halves the response time.

**Review by philipostli — COMMENTED** [review]
Seems legit! Not tested locally yet. Will try later.

Jitter only for slowPoll (every 7+-2 minutes) that gathers yearly energy. Tried to get 5000 pages all at once at same clockhours. I had no trouble with this, but it is good to filter out noise in the API and also have reliable data every 7 minutes. 
Should there be jitter on pollValues too? They are much more frequent.

Love the backoff logic.

Good work!

**Comment by PatrickE94** [comment]
Hello everyone!

Very nice to see some contributions here. This looks all nice and dandy to me. I've lost access to Zaptec devices so I'm unable to verify it myself locally. However, I can't see any issues with this :)

I'll merge and push!

File: drivers/go/device.ts (hunk 1 above)
Comment (thecoldwine) [reply drivers/go/device.ts:38]: Hi, jitter is helpful on every job that is running on the schedule, but it makes more difference on any request that potentially has big response body (just due to the memory constraints on the backend when you have too many in-flight requests).

We will have a look there after we inspect the workload from Homey more closely.

Thanks for the good work and merry Christmas!

**Pull request merged**